src/psyclone/dynamo0p3.py

```diff
@@ -2864,6 +4002,102 @@ def deallocate(self, parent):
             parent.add(DeallocateGen(parent, sorted(func_space_var_names)))
 
 
+class DynBoundaryConditions(DynCollection):
+    '''
+    Manages declarations and initialisation of quantities required by
+    kernels that need boundary condition information.
+
+    :param node: the Invoke or Kernel stub for which we are to handle \
+                 any boundary conditions.
+    :type node: :py:class:`psyclone.dynamo0p3.DynInvoke` or \
+                :py:class:`psyclone.dynamo0p3.DynKern`
+
+    :raises GenerationError: if a kernel named "enforce_bc_code" is found \
+                             but does not have an argument on ANY_SPACE_1.
+    :raises GenerationError: if a kernel named "enforce_operator_bc_code" is \
+                             found but does not have exactly one argument.
+    '''
+    # Define a BoundaryDofs namedtuple to help us manage the arrays that
+    # are required.
+    from collections import namedtuple
+    BoundaryDofs = namedtuple("BoundaryDofs", ["argument", "function_space"])
```

Comment:
Why is pycodestyle happy with BoundaryDofs? Shouldn't it be boundary_dofs?

Reply:
It's the name of a type so I think we're good.